ion-schema/src/model/type_reference.rs

```diff
@@ -13,17 +13,24 @@ use ion_rs::{Element, StructWriter, Value, ValueWriter};
 pub struct TypeReference {
     schema_id: Option<String>,
     type_name: String,
-    // Not exposed in public API.
-    // Initially `None`, but changed to `Some(_)` when resolving the schemas
-    // TODO: resolved: Cell<Option<...>>,
+    // Not exposed in public API. Initially `None`, but changed to `Some(_)` when resolving the schemas
+    resolved_type_coordinates: Option<TypeCoordinates>,
 }
 
 impl TypeReference {
-    // TODO: Determine if this needs to be pub
-    pub(crate) fn imported<A: Into<String>, B: Into<String>>(schema_id: A, type_name: B) -> Self {
+    fn new<S: Into<String>>(type_name: S) -> Self {
```

Comment:
🪧 There's now a public constructor for local type references (instead of just using `"foo".into()` or something like that).